rialto_airflow/harvest/distill.py

```diff
@@ -103,24 +96,112 @@ def _open_access(pub):
     )
 
 
+def _apc(pub, context):
+    """
+    Get the APC cost from one place in the openalex data , an external dataset, or another place in
+    OpenAlex data.
+    """
+    # https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/CR1MMV
+    return _first(
+        pub,
+        rules=[
+            JsonPathRule("openalex_json", "apc_paid.value_usd"),
+            FuncRule("openalex_json", _apc_oa_dataset, context),
+            JsonPathRule("openalex_json", "apc_list.value_usd"),
+        ],
+    )
+
+
+#
+# Helper functions to extract from specific JSON data structures that
+# aren't workable using JSON Path by itself.
+#
+
+
+def _wos_title(wos_json):
+    jsonp = jsonpath_ng.parse("static_data.summary.titles[*].title[*]")
+    for title in jsonp.find(wos_json):
+        if isinstance(title.value, dict) and title.value.get("type") == "item":
+            return title.value.get("content")
+
+    return None
+
+
+def _open_access_dim(dim_json):
+    """
+    Get the open access value, but ignore "oa_all"
+    """
+    jsonp = jsonpath_ng.parse("open_access[*]")
+    for oa in jsonp.find(dim_json):
+        if oa.value and oa.value != "oa_all":
+            return oa.value
+
+    return None
+
+
+def _apc_oa_dataset(openalex, context):
+    if openalex is None:
```

Comment:
Looks like you were able to switch this to work with OpenAlex since there are ISSNs there after all?

Reply:
Oh whoops, I just realized it's supposed to use the Dimensions ISSN not OpenAlex. I guess it should try to use both? I'll add an issue to take another look at this depending on what @jacobthill would like.